### PR TITLE
Elevate database engines in setup

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/18918

Adds special UI for the most popular database engines.

How to test:
- Run a fresh Metabase instance
- During setup it should be possible to search for and select a database
- Admin database forms should be unaffected 

Screen Shot 2021-11-11 at 8 37 43 pm
Screen Shot 2021-11-11 at 8 37 52 pm
Screen Shot 2021-11-11 at 8 38 01 pm
Screen Shot 2021-11-11 at 8 39 03 pm
Screen Shot 2021-11-11 at 8 38 08 pm
Screen Shot 2021-11-11 at 9 21 50 pm

### CODE REVIEW

**Review by alxnddr — CHANGES_REQUESTED** [review]
It looks there is a names mismatch. It tries to load the following SVGs but they are added with different names: `bigquery-cloud-sdk.svg` `presto-jdbc.svg`

Screen Shot 2021-11-15 at 13 36 17

Also, not related but must be easy to fix by adding the right url. I'm wondering if you can fix this link when the presto driver is selected:
Screen Shot 2021-11-15 at 13 39 28

**Comment by ranquild** [comment]
@alxnddr BE devs refused to add `elevated-engines` setting so I've moved the array to `lib/engines.js`. I've also fixed icons and link issues, please have a look.

**Review by alxnddr — APPROVED** [review]
👍

File: frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+};
+
+const EngineEmptyState = () => {
+  return (
+    <EngineEmptyStateRoot>
+      <EngineEmptyIcon name="search" size={32} />
+      <EngineEmptyText>{jt`Don’t see your database? Check out our ${(
+        <ExternalLink href="https://www.metabase.com/docs/latest/developers-guide-drivers.html">
+          {t`Community Drivers`}
+        </ExternalLink>
```
Comment (kulyk) [inline frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.jsx:148]: I suppose we should hide the Community Drivers link for cloud instances. AFAIK, community drivers are not currently supported there (I can be wrong though)

Comment (ranquild) [reply frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.jsx:148]: Good catch!

**Pull request merged**